libraries/chain/include/eosio/chain/trace.hpp

```diff
@@ -93,6 +93,32 @@ namespace eosio { namespace chain {
       return {action_id};
    }
 
+   #define DISK_EVENT_ID( FORMAT, ... ) \
+      fc::format_string( FORMAT, fc::mutable_variant_object()__VA_ARGS__ )
+
+   struct disk_trace {
```

Comment:
There doesn't seem to really be a need to create a `disk_trace` object when it is identical to `ram_trace`. Why not just rename `ram_trace` to `mem_usage_trace` and use it for both `RAM` and `DISK`?

Reply:
I agree they share similar information for a similar use case. I'm trying to find arguments to have two separate entity, apart from the fact that design wise, they make more sense, it's still two different kind of trace after all, I don't see much pros. 

But having a single structure make it harder to distinguish between RAM & Disk usage, but there is no use for this right now and surrounding context where they are going to be used for now will be enough to known which of RAM or Disk it targets.

The name `mem_usage_trace` is not super however, clearly a disk usage is not memory consumed. The name `storage_trace` or `storage_usage_trace` feels better, other suggestions: `data_usage_trace`, `bytes_usage_trace`, `resource_usage_trace`. I let you pick, or conserve `mem_usage_trace`.

Reply:
I like `storage_usage_trace`.